### PR TITLE
Extend backups for destinations

### LINKED ISSUE #7938: Expand EtcdBackupConfig with a destination field
**User Story**
To support multiple backup destinations per Seed, we need to expand the EtcdBackupConfig with Destination information, so the controller knows where to store the backups

### PR DESCRIPTION
**Pull request by lsviben**
**What this PR does / why we need it**:
Extends Seed with the option of multiple destinations for backups, and adds a destination field for EtcdBackupConfig. 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #7938 

**Does this PR introduce a user-facing change?**:

```release-note
seed.backupRestore is now deprecated in favor of seed.etcdBackupRestore which offers the option to manage multiple etcd backup destinations. The deprecated seed.backupRestore is still supported for now. 
```

### CODE REVIEW

**Review by moadqassem — APPROVED** [review]
/approve
/lgtm

File: pkg/apis/kubermatic/v1/datacenter.go [hunk 1]
```diff
@@ -98,19 +97,39 @@ type SeedSpec struct {
...
 	BackupRestore *SeedBackupRestoreConfiguration `json:"backupRestore,omitempty"`
+	// EtcdBackupRestore holds the configuration of the automatic etcd backup restores for the Seed
+	EtcdBackupRestore *EtcdBackupRestore `json:"etcdBackupRestore,omitempty"`
 }
 
-// SeedBackupRestoreConfiguration are s3 settings used for backups and restores of user cluster etcds.
+// SeedBackupRestoreConfiguration defines the bucket name and endpoint as a backup destination.
+// Deprecated: use EtcdBackupRestore
 type SeedBackupRestoreConfiguration struct {
-	// +kubebuilder:default=s3.amazonaws.com
```
Comment (kron4eg) [inline pkg/apis/kubermatic/v1/datacenter.go:106]: is it not defaults anymore?

File: pkg/apis/kubermatic/v1/datacenter.go [hunk 2]
```diff
@@ -98,19 +97,39 @@ type SeedSpec struct {
...
+	// it enables automatic backup and restore for the seed.
+	Destinations map[string]*BackupDestination `json:"destinations,omitempty"`
+}
+
+// BackupDestination defines the bucket name and endpoint as a backup destination, and holds reference to the credentials secret.
+type BackupDestination struct {
+	// Endpoint is the API endpoint to use for backup and restore.
+	Endpoint string `json:"endpoint,omitempty"`
+	// BucketName is the bucket name to use for backup and restore.
+	BucketName string `json:"bucketName,omitempty"`
```
Comment (kron4eg) [inline pkg/apis/kubermatic/v1/datacenter.go]: we can default Endpoint, but do we default BucketName? I mean, `,omitempty` here means 'not-required field' in terms of generated OpenAPI schema. Is this what we want to allow?

File: pkg/apis/kubermatic/v1/datacenter.go (hunk 1 above)
Comment (lsviben) [reply pkg/apis/kubermatic/v1/datacenter.go:106]: It never defaulted it anywhere, the comment "defaults to s3.amazonaws.com" was misleading unfortunately.

File: pkg/apis/kubermatic/v1/datacenter.go (hunk 2 above)
Comment (lsviben) [reply pkg/apis/kubermatic/v1/datacenter.go]: good point, but this messes up our PATCH endpoints as the fields which are not in the patch without omitempty get defaulted to "", and overwrite the old value. 

But I can easily create an API model to solve that, so Ill change it

**Review by kron4eg — APPROVED** [review]
/approve

**Pull request merged**